### PR TITLE
[IE CLDNN] Add common comparator for ngraph reference and cldnn detection output primitive

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Add common comparator for ngraph reference and cldnn detection output primitive
 - Sort function to be aligned with others(cldnn/mkldnn detection output primitive)

### Tickets:
 - 45293

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp [hunk 1]
```diff
@@ -184,7 +184,11 @@ struct detection_output_cpu : typed_primitive_impl<detection_output> {
                    const bool share_location,
                    std::map<int, std::vector<int>>& indices,
                    std::vector<std::pair<float, std::pair<int, int>>>& scoreIndexPairs) {
-        std::sort(scoreIndexPairs.begin(), scoreIndexPairs.end(), comp_score_descend<std::pair<int, int>>);
+        std::sort(scoreIndexPairs.begin(),
+                  scoreIndexPairs.end(),
+                  [](const std::pair<float, std::pair<int, int>>& p1, const std::pair<float, std::pair<int, int>>& p2) {
+                      return (p1.first > p2.first) || (p1.first == p2.first && p1.second.second < p2.second.second);
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp:190]: comp_score_descend is fixed according to this lambda function, but why not using that?

File: inference-engine/src/mkldnn_plugin/nodes/detectionoutput.cpp [hunk 2]
```diff
@@ -263,8 +263,11 @@ class DetectionOutputImpl: public ExtLayerBase {
                     }
                 }
 
-                std::sort(conf_index_class_map.begin(), conf_index_class_map.end(),
-                          SortScorePairDescend<std::pair<int, int>>);
+                std::sort(conf_index_class_map.begin(),
+                          conf_index_class_map.end(),
+                          [](const std::pair<float, std::pair<int, int>>& p1, const std::pair<float, std::pair<int, int>>& p2) {
+                              return (p1.first > p2.first) || (p1.first == p2.first && p1.second.second < p2.second.second);
+                          });
```
Comment (yeonbok) [inline inference-engine/src/mkldnn_plugin/nodes/detectionoutput.cpp:270]: why not using SortScorePairDescend ? it is already fixed aligned to this

File: inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp (hunk 1 above)
Comment (andrew-k-park) [reply inference-engine/thirdparty/clDNN/src/gpu/detection_output_cpu.cpp]: fixed

File: inference-engine/src/mkldnn_plugin/nodes/detectionoutput.cpp (hunk 2 above)
Comment (andrew-k-park) [reply inference-engine/src/mkldnn_plugin/nodes/detectionoutput.cpp]: fixed

**Comment by yeonbok** [comment]
Looks good to me

**Comment by yeonbok** [comment]
@vladimir-paramuzov Ping, please review this.

**Review by ilyachur — COMMENTED** [review]
Could you please add new test for the fix?

**Comment by yeonbok** [comment]
> Could you please add new test for the fix?

I am not sure it makes sense to add a new test, because currently detection output layer is been implemented only in CPU and all of them uses same std::sort. When gpu implementation comes (which we are preparing now and will be added in a near future), which will use a custom implementation of sort, and at that time this custom comparator will play an important role to sync b/w cpu and gpu.

**Pull request closed without merging**